### ISSUE TITLE
Importing ufl before firedrake wildcard makes mumps fail to initialize

### ISSUE THREAD

**Issue by salazardetroya** [firedrakeproject/firedrake#1800]
```python
import ufl
from firedrake import *

mesh = UnitSquareMesh(10, 10)
V = FunctionSpace(mesh, 'CG', 1)
v = TestFunction(V)
u = TrialFunction(V)
a = inner(grad(u), grad(v)) * dx
L = - 1.0 * v * dx
bc = DirichletBC(V, 0.0, "on_boundary")

parameters = {
        "ksp_type": "preonly",
        "pc_type": "lu",
        "pc_factor_mat_solver_type": "mumps",
        }

u_sol = Function(V)
solve(a==L, u_sol, solver_parameters=parameters)
```
causes
```
[0]PETSC ERROR: ------------------------------------------------------------------------
[0]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of range
[0]PETSC ERROR: Try option -start_in_debugger or -on_error_attach_debugger
[0]PETSC ERROR: or see https://www.mcs.anl.gov/petsc/documentation/faq.html#valgrind
[0]PETSC ERROR: or try http://valgrind.org on GNU/linux and Apple Mac OS X to find memory corruption errors
[0]PETSC ERROR: likely location of problem given in stack below
[0]PETSC ERROR: ---------------------  Stack Frames ------------------------------------
[0]PETSC ERROR: Note: The EXACT line numbers in the stack are not available,
[0]PETSC ERROR:       INSTEAD the line number of the start of the function
[0]PETSC ERROR:       is given.
[0]PETSC ERROR: [0] PetscInitializeMUMPS line 1849 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/mat/impls/aij/mpi/mumps/mumps.c
[0]PETSC ERROR: [0] MatGetFactor_aij_mumps line 2985 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/mat/impls/aij/mpi/mumps/mumps.c
[0]PETSC ERROR: [0] MatGetFactor line 4511 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/mat/interface/matrix.c
[0]PETSC ERROR: [0] PCSetUp_LU line 50 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/ksp/pc/impls/factor/lu/lu.c
[0]PETSC ERROR: [0] PCSetUp line 921 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/ksp/pc/interface/precon.c
[0]PETSC ERROR: [0] KSPSetUp line 318 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c
[0]PETSC ERROR: [0] KSPSolve_Private line 619 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c
[0]PETSC ERROR: [0] KSPSolve line 877 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c
[0]PETSC ERROR: [0] SNESSolve_KSPONLY line 14 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/snes/impls/ksponly/ksponly.c
[0]PETSC ERROR: [0] SNESSolve line 4399 /Users/salazardetro1/scicomp_libraries/firedrake-debug/firedrake/src/petsc/src/snes/interface/snes.c
application called MPI_Abort(MPI_COMM_WORLD, 50151059) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=50151059
:
system msg for write_line failure : Bad file descriptor
```
Comment `import ufl` to make it work.

**Comment by wence-** [comment]
Thanks, I think this is this numpy gfortran issue.

**Comment by wence-** [comment]
What platform are you on? I can reproduce on MacOS 10.14 (Mojave).

It goes away if I explicitly `LD_PRELOAD` the right libgfortran via `DYLD_INSERT_LIBRARIES`.

**Comment by salazardetroya** [comment]
I am on MacOS 10.14 (Mojave) as well.

**Comment by wence-** [comment]
Fixed by #1885, needs a fresh install with a newly download `firedrake-install` script.
